### ISSUE TITLE
Is there a way to stop arguments from being interpreted as flags?

### ISSUE THREAD

**Issue by Mrfiregem** [hellflame/argparse#22]
Most cli programs allow specifying `--` to make it so any arguments to the right of it are not treated as flags, even if they take the form of `-a` or `--arg`. Is there any way to implement this currently with argparse? For example:

```bash
# '--arg2' should be treated as a positional argument after '--'
./my_command -ab --arg2 -- arg1 --arg2 arg3
```

**Comment by hellflame** [comment]
Hi there, can you give me a specific use case of this feature? or the program that provide this feature and the way you use this --

**Comment by Mrfiregem** [comment]
Sure. Most standard Linux or GNU programs like `cat`, `ls`, `rm`, etc. support using -- to turn off flag parsing for any argument after it. For example, on my Mac computer:

```bash
bash-3.2$ echo 'hello' > --text_file
bash-3.2$ cat --text_file 
cat: illegal option -- -
usage: cat [-belnstuv] [file ...]
bash-3.2$ cat -- --text_file 
hello
```

For my specific use case, I'm writing a program that calls another program, and if I want to pass any flags to the 2nd program, I could put them after -- so my Go program with your package wouldn't error out or erroniously treat them as its own flags. E.g. `go run main.go --flag1_for_go_program ls --flag2_for_go_program positional_argument -- --flag_for_ls`

**Comment by hellflame** [comment]
I see. Arguments after `--` will be treated as positional. This is quite reasonable, I think I'll add this feature soon. 
Thanks for the advice!

**Comment by hellflame** [comment]
I see. Arguments after -- will be treated as positional. This is quite reasonable, I think I'll add this feature soon. 
Thanks for the advice!

                       
Original
                       
                     

From:"Firegem"< ***@***.*** &gt;;

Date:2023/11/18 6:27

To:"hellflame/argparse"< ***@***.*** &gt;;

CC:"hellflame"< ***@***.*** &gt;;"Assign"< ***@***.*** &gt;;

Subject:Re: [hellflame/argparse] Is there a way to stop arguments from beinginterpreted as flags? (Issue #22)




 
Sure. Most standard Linux or GNU programs like cat, ls, rm, etc. support using -- to turn off flag parsing for any argument after it. For example, on my Mac computer:
 bash-3.2$ echo 'hello' &gt; --text_file bash-3.2$ cat --text_file  cat: illegal option -- - usage: cat [-belnstuv] [file ...] bash-3.2$ cat -- --text_file  hello
 
For my specific use case, I'm writing a program that calls another program, and if I want to pass any flags to the 2nd program, I could put them after -- so my Go program with your package wouldn't error out or erroniously treat them as its own flags. E.g. go run main.go --flag1_for_go_program ls --flag2_for_go_program positional_argument -- --flag_for_ls

**Comment by hellflame** [comment]
Hi there. The feature is published on v1.12. You can have some try now. 
If there's more requests or questions, feel free to open an issue. Thanks for the advice again.

**Comment by Mrfiregem** [comment]
Thank you for the quick implementation, looks great!
